File: src/NnCase.IR/Operators/Softmax.cs
using System;
using System.Collections.Generic;
using System.Numerics.Tensors;
using System.Text;

namespace NnCase.IR.Operators
{
    public class Softmax : Node
    {
        public InputConnector Input { get; }

        public OutputConnector Output { get; }

        public float Beta { get; }

        public Softmax(Shape inputShape, float beta)
        {
            Beta = beta;

            Input = AddInput("input", DataType.Float32, inputShape);
            Output = AddOutput("output", DataType.Float32, inputShape);
        }
    }
}

File: src/ir/ops/softmax.cpp
#include <ir/ops/softmax.h>
#include <ir/op_utils.h>
#include <xtensor/xarray.hpp>

using namespace nncase;
using namespace nncase::ir;

softmax::softmax(shape_t input_shape, float beta)
    : beta_(beta)
{
    add_input("input", dt_float32, input_shape);
    add_output("output", dt_float32, input_shape);
}
